parmed/structure.py

```diff
@@ -1119,16 +1121,14 @@ def __getitem__(self, selection):
         scan = [x * y for x, y in zip(scan, selection)]
         # Copy all parameters
         struct = type(self)()
+        res_numbers_are_fake = {res.number for res in self.residues} == {0}
```

Comment:
I actually think they are `-1` if they are fake. And if you do that here, then you might want to use it in `__repr__`, too.

Reply:
-1 is the (stupid) not-set default.  The existing logic was only checking if they were 0, which leads me to think that they were set to 0 in one of the parsers.

Reply:
necrocommenting here, but -1 is sometimes a valid resid (if residues have been inserted before the canonical 0).  Depending on the source of this (if this is internal book-keeping and you're sure it's never -1) then you might be better off using a safer sentinel value (-MAX_INT or something)